packages/web/pages/highlights.tsx

```diff
@@ -0,0 +1,352 @@
+import { showErrorToast, showSuccessToast } from '../lib/toastHelpers'
+
+const PAGE_SIZE = 10
+
+export default function HighlightsPage(): JSX.Element {
+  const router = useRouter()
+  const viewer = useGetViewerQuery()
+  const [showFilterMenu, setShowFilterMenu] = useState(false)
+  const [_, setShowAddLinkModal] = useState(false)
+
+  useApplyLocalTheme()
+
+  const { isLoading, setSize, size, data, mutate } = useGetHighlights({
+    first: PAGE_SIZE,
+  })
+
+  const hasMore = useMemo(() => {
+    if (!data) {
+      return false
+    }
+    return data[data.length - 1].highlights.pageInfo.hasNextPage
+  }, [data])
+
+  const handleFetchMore = useCallback(() => {
+    if (isLoading || !hasMore) {
+      return
+    }
+    setSize(size + 1)
+  }, [isLoading, hasMore, setSize, size])
+
+  useFetchMore(handleFetchMore)
+
+  const highlights = useMemo(() => {
+    if (!data) {
+      return []
+    }
+    return data.flatMap((res) => res.highlights.edges.map((edge) => edge.node))
+  }, [data])
+
+  if (!highlights.length) {
+    return (
+      <Box
+        css={{
+          width: '100%',
+          height: `100vh`,
+        }}
+      >
+        <EmptyHighlights />
+      </Box>
+    )
+  }
+
+  return (
+    <HStack>
+      <Toaster />
+
+      <LibraryFilterMenu
+        setShowAddLinkModal={setShowAddLinkModal}
+        showFilterMenu={showFilterMenu}
+        setShowFilterMenu={setShowFilterMenu}
+        searchTerm={undefined}
+        applySearchQuery={(searchQuery: string) => {
+          router?.push(`/home?q=${searchQuery}`)
+        }}
+      />
+      <VStack
+        css={{
+          maxWidth: '70%',
+          padding: '20px',
+          margin: '30px 50px 0 0',
+        }}
+      >
+        {highlights.map((highlight) => {
+          return (
+            viewer.viewerData?.me && (
+              <HighlightCard
+                key={highlight.id}
+                highlight={highlight}
+                viewer={viewer.viewerData.me}
+                router={router}
+                mutate={mutate}
+              />
+            )
+          )
+        })}
+      </VStack>
+    </HStack>
+  )
+}
+
+type HighlightCardProps = {
+  highlight: Highlight
+  viewer: UserBasicData
+  router: NextRouter
+  mutate: () => void
+}
+
+type HighlightAnnotationProps = {
+  highlight: Highlight
+}
+
+function HighlightAnnotation({
+  highlight,
+}: HighlightAnnotationProps): JSX.Element {
+  const [noteMode, setNoteMode] = useState<'edit' | 'preview'>('preview')
+  const [annotation, setAnnotation] = useState(highlight.annotation)
+
+  return (
+    <HighlightViewNote
+      targetId={highlight.id}
+      text={annotation}
+      placeHolder="Add notes to this highlight..."
+      highlight={highlight}
+      mode={noteMode}
+      setEditMode={setNoteMode}
+      updateHighlight={(highlight) => {
+        setAnnotation(highlight.annotation)
+      }}
+    />
+  )
+}
+
+function HighlightCard(props: HighlightCardProps): JSX.Element {
```

Comment:
It doesn't matter now, but once merged I will pull this into its own file when i rebase mine.